www/pfSense-pkg-squid/files/usr/local/pkg/squid.inc

```diff
@@ -1917,7 +1917,7 @@ function squid_resync_auth() {
 			case 'ldap':
 				$port = ((isset($settings['auth_server_port']) && !empty($settings['auth_server_port'])) ? ":{$settings['auth_server_port']}" : '');
 				$password = (isset($settings['ldap_pass']) ? "-w {$settings['ldap_pass']}" : '');
-				$conf .= "auth_param basic program " . SQUID_LOCALBASE . "/libexec/squid/basic_ldap_auth -v {$settings['ldap_version']} -b {$settings['ldap_basedomain']} -D {$settings['ldap_user']} $password -f \"{$settings['ldap_filter']}\" -u {$settings['ldap_userattribute']} -P {$settings['auth_server']}$port\n";
+				$conf .= "auth_param basic program " . SQUID_LOCALBASE . "/libexec/squid/basic_ldap_auth -v {$settings['ldap_version']} -b " . escapeshellarg($settings['ldap_basedomain']) . " -D " . escapeshellarg($settings['ldap_user']) . " " . escapeshellarg($password) . " -f " . escapeshellarg($settings['ldap_filter']) . " -u " . escapeshellarg($settings['ldap_userattribute']) . " -P " . escapeshellarg($settings['auth_server']) . escapeshellarg($port) . "\n";
```

Comment:
The way `$port` is formed the address and port should be one string, so this bit:
`escapeshellarg($settings['auth_server']) . escapeshellarg($port)`
Should be:
`escapeshellarg($settings['auth_server'] . $port)`

Reply:
Also it would be easier to read if this was split into multiple lines. A newline and some tab intents for each command line parameter would make it look a lot better.

Reply:
done